pkg/redfish/redfish-foxconn.go

```diff
@@ -0,0 +1,323 @@
+// MIT License
+//
+// (C) Copyright [2024] Hewlett Packard Enterprise Development LP
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+// OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+package rf
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+const FOXCONN_ETH_INT_DESCRIPTION     = "Foxconn NCSI Interface"
+const FOXCONN_PRIMARY_ETH_INT_SUFFIX  = "-primary_eth"
+const FOXCONN_PRIMARY_ETH_INT_PCIDID  = "0x6315"
+const FOXCONN_PRIMARY_ETH_INT_FW_NAME = "X550 FW Ver"
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// localhost:~ # curl -ks -u root:${BMC_CREDS} https://10.5.1.106/redfish/v1/Systems/system | jq .
+// {
+//     "@odata.context": "",
+//     "@odata.id": "/redfish/v1/Systems/system",
+//     "@odata.type": "#ComputerSystem.v1_18_0.ComputerSystem",
+//     ... <REDACTED> ...
+//     "Oem": {
+//         "InsydeNcsi": {
+//             "Ncsi": {
+//                 "@odata.id": "/redfish/v1/Systems/system/Oem/Insyde/Ncsi"
+//             }
+//         }
+//     }
+//     ... <REDACTED> ...
+// }
+//
+///////////////////////////////////////////////////////////////////////////////
```

Comment:
I know these comment blocks might seem like a lot to include in the source code but they were instrumental in understanding how to construct the data structures necessary to parse the redfish endpoints.